_search-plugins/search-relevance/ml-inference-rerank-by-field.md

```diff
@@ -0,0 +1,277 @@
+{ "index": { "_id": 6 } }
+{ "borough": "Staten Island", "area_name": "St. George", "description": "St. George is the main commercial and cultural center of Staten Island, offering stunning views of Lower Manhattan.", "population": 15000, "facts": "St. George is home to the Staten Island Ferry terminal and is a gateway to Staten Island, offering stunning views of the Statue of Liberty and Ellis Island." }
+
+```
+{% include copy-curl.html %}
+
+## Step 2: Create a connector
+Create a conector assuming you have created a sagemaker model with a cross encoder 
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "SageMaker cross-encoder model",
+  "description": "Test connector for Sagemaker cross-encoder model",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+		"access_key": "<Access key>",
+		"secret_key": "<Secret key>",
+		"session_token": "<Session token>"
+  },
+  "parameters": {
+    "region": "us-east-1",
+    "service_name": "sagemaker"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "<SageMaker Endpoint>",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "request_body": "{ \"inputs\":${parameters.inputs}}" ,
+       "pre_process_function": 
+      """
+        def json = params.json;
+        def inputs = json.parameters.inputs;
+        
+        def result = [:];
+        result.query_text = inputs.text;
+        result.text_docs = [inputs.text_pair];
+        
+        return result;
+      """
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+
+## Step 3: Create a model
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "text classification model",
+  "version": "1.0.1",
+  "function_name": "remote",
+  "description": "Text Classification",
+  "connector_id": "<connector_id_from_step_2>"
+} 
+
+```
+{% include copy-curl.html %}
+
+
+## Step 4: Create the search pipeline
+Now we will apply our search pipeline our input will use the facts from each document. and the text pair will come from our initial query
+to the documents we want to compare with.
+
+As for reranking; the data returned will be `rank_score` which will be used to reorder the documents based on the data the cross encoder provided. We will to remove the `rank_score`
+field as retaining will show the same value for the `_score` field and the `_source.rank_score` field. We will also keep the previous score metric before the reranking for debugging purposes
+and for emphasis of improvement of search relevancy according the search query.
+
+```json
+PUT /_search/pipeline/my_pipeline
+{
+  "response_processors": [
+    {
+      "ml_inference": {
+        "tag": "ml_inference",
+        "description": "This processor runs ml inference during search response",
+        "model_id": "<model_id_from_step_3>",
+        "model_input":"""{"parameters":{"inputs":{"text":"${input_map.text}","text_pair":"${input_map.text_pair}"}}}""",
```

Comment:
How about move this to connector configuration?

Reply:
Apologies for the delay I have moved this to the connector in the request_body of creating the connector